### DISCUSSION TITLE
Panel name empty in canAccessPanel

### DISCUSSION THREAD

**Discussion by azizultex** [filamentphp/filament#8602] [Help] [answered]
Hello everyone, 
I have 2 user roles admin and customers in the dashboard I'm working on. I created a panel App for customer role only. And restrict the Customers to access Admin panel. 

I was following Filament documentation to control the access but I'm not getting the $panel. It shoes empty array. 

```
public function canAccessPanel(Panel $panel): bool 
{
    Log::info('User panel: ', ['panel' => $panel]);

    // return $this->role == 'customer' && $panel == 'app';
    return true;
}
```

I find this in the log: 
 `local.INFO: User panel:  {"panel":{"Filament\\Panel":[]}} `
 
 What am I missing here? Any help highly appreciated.

**Comment by aymanalareqi** [comment, marked as answer]
the $panel is object of Panel Class so, you should could $panel->getId() to access panel id like this
```
public function canAccessPanel(Panel $panel): bool 
{ 
    return $this->role == 'customer' && $panel->getId() == 'app';
}

**Reply by azizultex** [reply]
This works! Thanks
